controller/src/main/java/org/jboss/as/controller/ModelControllerLock.java

```diff
@@ -102,6 +144,28 @@ protected synchronized boolean tryAcquire(int permit) {
         }
 
         @Override
+        protected synchronized int tryAcquireShared(int permit) {
```

Comment:
The class was this way before, but its sort of strange to see an AQS with synced methods. If the access to the "owner" which IIUC is what "permitHolder" represents is done properly, then the sync is unnecessary. If there is some special case where an atomic update of both is required AbstractQueuedLongSynchronizer using a bit mask might be a better solution. This is mostly a quibble though, and could be addressed in a subsequent update.